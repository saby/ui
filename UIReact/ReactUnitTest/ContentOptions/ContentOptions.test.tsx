/**
 * @jest-environment jsdom
 */
import { render } from 'react-dom';
import { act } from 'react-dom/test-utils';

import IndexControl from 'ReactUnitTest/ContentOptions/resources/Index';

describe('Контентные опции', () => {
    let container: HTMLDivElement;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);

        jest.useFakeTimers();
    });

    afterEach(() => {
        jest.useRealTimers();
        container.remove();
    });

    /**
     * Cценарий, когда в контрол в опциях передается другой контрол и опции для него.<br/>
     * При этом после _forceUpdate родителя эти опции должны остаться теми же по ссылке.
     */
    it('Конетная опция не поменялась после _forceUpdate', () => {
        let instance;
        act(() => {
            render(
                <IndexControl
                    ref={(v) => {
                        instance = v;
                    }}
                />,
                container
            );
        });

        expect(container).toMatchSnapshot('1.построилось');

        act(() => {
            instance._children.parent._forceUpdate();
            jest.runAllTimers();
        });

        expect(container).toMatchSnapshot('2.после _forceUpdate');
    });
});
