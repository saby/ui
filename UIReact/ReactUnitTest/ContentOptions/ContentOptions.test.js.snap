// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Контентные опции Конетная опция не поменялась после _forceUpdate: 1.построилось 1`] = `
<div>
  <div
    tabindex="0"
  >
    <div
      tabindex="0"
    >
      Содержимое "ребенка", в который передали опции объектом из wml-шаблона родителя
    </div>
    <div>
      Опции остались те же - 
    </div>
  </div>
</div>
`;

exports[`Контентные опции Конетная опция не поменялась после _forceUpdate: 2.после _forceUpdate 1`] = `
<div>
  <div
    tabindex="0"
  >
    <div
      tabindex="0"
    >
      Содержимое "ребенка", в который передали опции объектом из wml-шаблона родителя
    </div>
    <div>
      Опции остались те же - ДА
    </div>
  </div>
</div>
`;
